secure/storage/src/vault.rs

```diff
@@ -78,13 +78,11 @@ impl Storage for VaultStorage {
         // Vault internally does not distinguish creation versus update except by permissions. So we
         // simulate that by first getting the key. If it doesn't exist, we're okay and return back a
         // fake, but ignored value.
-        self.get_secret(&key).or_else(|e| {
-            if let Error::KeyNotSet(_) = e {
-                Ok(Value::U64(0))
-            } else {
-                Err(e)
-            }
-        })?;
+        match self.get_secret(&key) {
+            Ok(_) => return Err(Error::KeyAlreadyExists(key.to_string())),
+            Err(Error::KeyNotSet(_)) => (/* Expected this for new keys! */),
```

Comment:
fix the comment above and/or delete this comment.

good find, would recommend that we put this commit first and maybe in another PR to move fast

Reply:
Done, thanks! I've separated it into a different PR and sent that across to you.